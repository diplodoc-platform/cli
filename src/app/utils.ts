import {TextSizes, Theme} from '@doc-tools/components';

const DEFAULT_USER_SETTINGS = {
    theme: Theme.Light,
    textSize: TextSizes.m,
    showMiniToc: true,
    wideFormat: true,
    fullScreen: false,
};

export function getDocSettings() {
    const {
        theme: defaultTheme,
        textSize: defaultTextSize,
        showMiniToc: defaultShowMiniToc,
        wideFormat: defaultWideFormat,
        fullScreen: defaultFullScreen,
    } = DEFAULT_USER_SETTINGS;

    const theme = getSetting('theme') as Theme || defaultTheme;
    const textSize = getSetting('textSize') as TextSizes || defaultTextSize;
    const showMiniToc = getSetting('showMiniToc') || defaultShowMiniToc;
    const wideFormat = getSetting('wideFormat') || defaultWideFormat;
    const fullScreen = getSetting('fullScreen') || defaultFullScreen;

    return {
        theme,
        textSize,
        showMiniToc: strToBoolean(showMiniToc),
        wideFormat: strToBoolean(wideFormat),
        fullScreen: strToBoolean(fullScreen),
    };
}

export function getSetting(name: string) {
    return sessionStorage.getItem(name);
}

export function saveSetting<T>(name: string, value: T) {
    sessionStorage.setItem(name, String(value));
}

export function strToBoolean(str: any) {
    if (typeof str === 'boolean') {
        return str;
    }

    return str ? str === 'true' : false;
}

export function withSavingSetting<T>(settingName: string, onChange: (value: T) => void) {
    return (value: T) => {
        saveSetting<T>(settingName, value);

        onChange(value);
    };
}

export function updateRootClassName(theme: Theme, isMobile = false) {
    const themeClassName = theme === 'light' ? 'yc-root_theme_light' : 'yc-root_theme_dark';
    const mobileClassName = isMobile ? 'mobile' : '';

    document.body.className = `yc-root ${themeClassName} ${mobileClassName}`;
}
