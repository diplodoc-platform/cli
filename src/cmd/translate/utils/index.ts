import {ok} from 'assert';
import {basename, dirname, extname, resolve} from 'path';
import {readFileSync} from 'node:fs';
import glob from 'glob';

export {dumpFile, loadFile, resolveSchemas} from './fs';
export {extract, compose} from './translate';
export {TranslateError, LimitExceed, RequestError, AuthError} from './errors';

type TranslateArgs = {
    input: string;
    output?: string;
    source?: string;
    sourceLanguage?: string;
    sourceLanguageLocale?: string;
    target?: string;
    targetLanguage?: string;
    targetLanguageLocale?: string;
    auth?: string;
    folder?: string;
    glossary?: string;
    include?: string[] | string;
    exclude?: string[] | string;
    dryRun?: boolean;
    useSource?: boolean;
} & {
    [prop: string]: any;
};

export type TranslateParams = {
    input: string;
    output: string;
    source: [string, string];
    targets: [string, string][];
    auth?: string;
    folder?: string;
    glossary?: string;
    files: string[];
    dryRun: boolean;
    useSource: boolean;
};

export function normalizeParams(params: TranslateArgs, exts = EXTS): TranslateParams {
    const source = normalizeLocale(params, 'source')[0];
    const targets = normalizeLocale(params, 'target');
    const {input, files} = normalizeInput(params, source[0], exts);
    const {output, auth, folder, dryRun, useSource} = params;

    ok(input, 'Required param input is not configured');

    return {
        input,
        output: output || input,
        auth,
        folder,
        source,
        targets,
        files: [...new Set(files)],
        dryRun: Boolean(dryRun),
        useSource: Boolean(useSource),
    };
}

function normalizeLocale(params: TranslateArgs, scope: 'source' | 'target'): [string, string][] {
    const _scopeLanguage = params[scope + 'Language'];
    const _scopeLanguageLocale = params[scope + 'LanguageLocale'];
    const _scope = params[scope] || _scopeLanguageLocale || _scopeLanguage;

    if (!_scope) {
        return [['', '']];
    }

    if (Array.isArray(_scope)) {
        return _scope.map((_scope) => _scope.split('-'));
    }

    return [_scope.split('-') as [string, string]];
}

const EXTS = ['.md', '.yaml', '.json'];

function normalizeInput(params: TranslateArgs, language: string, exts: string[]) {
    let {input, include = [], exclude = []} = params;

    include = ([] as string[]).concat(include);
    exclude = ([] as string[]).concat(exclude);

    let files: string[] | null = null;
    if (extname(input) === '.list') {
        const list = readFileSync(input, 'utf8').split('\n');
        input = dirname(input);
        files = list.map((file) => {
            const absPath = resolve(input, file);

            if (!absPath.startsWith(input)) {
                throw new Error(`Insecure access to file out of project scope. (file: ${absPath})`);
            }

            if (!exts.includes(extname(file))) {
                throw new Error(`Unhandles file extension. (file: ${absPath})`);
            }

            return absPath;
        });
    } else {
        if (exts.includes(extname(input))) {
            files = [basename(input)];
            input = dirname(input);
        }

        if (!include.length) {
            include.push('...');
        }

        include = include.reduce((acc, item) => {
            if (item === '...') {
                acc.push(...exts.map((ext) => (language || '.') + '/**/*' + ext));
            } else {
                acc.push(item);
            }

            return acc;
        }, [] as string[]);

        files =
            files ||
            ([] as string[]).concat(
                ...include.map((match) =>
                    glob.sync(match, {
                        cwd: input,
                        ignore: exclude,
                        nodir: true,
                    }),
                ),
            );
    }

    return {input, files};
}

export class Defer<T = string> {
    resolve!: (text: T) => void;

    reject!: (error: any) => void;

    promise: Promise<T>;

    constructor() {
        this.promise = new Promise((resolve, reject) => {
            this.resolve = resolve;
            this.reject = reject;
        });
    }
}

export function bytes(texts: string[]) {
    return texts.reduce((sum, text) => sum + text.length, 0);
}
