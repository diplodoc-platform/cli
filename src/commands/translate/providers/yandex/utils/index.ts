export {LimitExceed, RequestError, AuthError} from './errors';

export class Defer<T = string> {
    resolve!: (text: T) => void;

    reject!: (error: any) => void;

    promise: Promise<T>;

    constructor() {
        this.promise = new Promise((resolve, reject) => {
            this.resolve = resolve;
            this.reject = reject;
        });
    }
}

export function bytes(texts: string[]) {
    return texts.reduce((sum, text) => sum + text.length, 0);
}
