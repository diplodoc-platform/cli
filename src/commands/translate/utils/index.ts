export type {Locale} from './config';
export {resolveSchemas, FileLoader} from './fs';
export {extract, compose} from './translate';
export {resolveSource, resolveTargets, resolveFiles, resolveVars} from './config';
export {
    TranslateError,
    ExtractError,
    ComposeError,
    SkipTranslation,
    EmptyTokensError,
} from './errors';
