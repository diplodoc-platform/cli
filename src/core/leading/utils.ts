import {cloneDeepWith, isString} from 'lodash';

export function modifyValuesByKeys(
    object: object,
    keys: string[],
    modify: (value: string) => string,
) {
    // Clone the object deeply with a customizer function that modifies matching keys
    return cloneDeepWith(object, (value: unknown, key) => {
        if (keys.includes(key as string) && isString(value)) {
            return modify(value);
        }

        return undefined;
    });
}
