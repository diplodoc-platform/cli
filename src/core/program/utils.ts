import type {BaseArgs, BaseConfig} from '~/core/program/types';
import type {BaseProgram} from '~/core/program/index';

import {isAbsolute} from 'node:path';

export function isRelative(path: string | undefined) {
    return /^\.{1,2}\//.test(path || '');
}

export function requireExtension(name: string) {
    if (isRelative(name) || isAbsolute(name)) {
        return require(name);
    }

    const local = (name: string) => {
        if (name.startsWith('@')) {
            return undefined;
        }

        return '@diplodoc/cli/' + name;
    };

    const scoped = (name: string, fixed = false) => {
        if (name.startsWith('@')) {
            return undefined;
        }

        return '@diplodoc/' + name + (fixed ? '-extension' : '');
    };

    const precise = (name: string, fixed = false) => {
        return name + (fixed ? '-extension' : '');
    };

    const variants = [
        local(name),
        scoped(name),
        scoped(name, true),
        precise(name),
        precise(name, true),
    ].filter(Boolean);

    while (variants.length) {
        const variant = variants.shift() as string;
        try {
            return require(variant);
        } catch {}
    }

    throw new Error(`Unable to resolve '${name}' extension`);
}

export function isProgram<TConfig extends BaseConfig, TArgs extends BaseArgs>(
    module: unknown,
): module is BaseProgram<TConfig, TArgs> {
    return Boolean(module && typeof (module as BaseProgram).init === 'function');
}
