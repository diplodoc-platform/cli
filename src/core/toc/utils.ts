import type {TextFilter} from './types';

import {evaluate} from '@diplodoc/liquid';

export function isRelative(path: AnyPath): path is RelativePath {
    return /^\.{1,2}\//.test(path) || !/^(\w{0,7}:)?\/\//.test(path);
}

export function getFirstValuable<T>(
    items: TextFilter[] | string,
    vars: Hash,
    fallback?: T,
): T | undefined {
    if (typeof items === 'string') {
        items = [{text: items, when: true}];
    }

    if (!Array.isArray(items)) {
        items = [];
    }

    for (const item of items) {
        let {when = true} = item;
        delete item.when;

        if (typeof when === 'string') {
            when = Boolean(evaluate(when, vars));
        }

        if (when) {
            return item.text as T;
        }
    }

    return fallback;
}
