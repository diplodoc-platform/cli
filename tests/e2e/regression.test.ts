import {dedent} from 'ts-dedent';
import {compareDirectories, getTestPaths, runYfmDocs} from '../utils';

function test(_description: string) {
    it('internal', () => {
        const {inputPath, outputPath} = getTestPaths(
            'mocks/regression',
        );

        runYfmDocs(inputPath, outputPath, {md2md: true, md2html: false});
        runYfmDocs(outputPath, outputPath + '-html', {
            md2md: false,
            md2html: true,
        });
        runYfmDocs(outputPath, outputPath + '-static-html', {
            md2md: false,
            md2html: true,
            args: '--static-content',
        });
        compareDirectories(outputPath);
        compareDirectories(outputPath + '-html');
        // compareDirectories(outputPath + '-static-html');
    });
}

describe('Regression', () => {
    test(dedent`
        - not_var liquid syntax
        - normalize leading hrefs
        - empty href in toc item
    `);
});
