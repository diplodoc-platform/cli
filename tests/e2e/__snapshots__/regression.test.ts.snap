// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression internal: 1.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Title
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{},"assets":[],"headings":[],"title":"Title","includes":[],"html":"
      <p>Text {{not-var}} {{skip()}} default-var-value
      </p>/n
      <p>Text
      <code>{{not-var}}
      </code>
      <code>{{skip()}}
      </code>
      <code>default-var-value
      </code>
      </p>/n
      <p>Text
      </p>/n/n
      <div class=\\"yfm-clipboard\\">/n
      <pre>
      <code class=\\"hljs\\">{{not-var}}/n/n{{skip()}}/n/ndefault-var-value/n
      </code>
      </pre>/n/n
      <button class=\\"yfm-clipboard-button\\">/n
      <svg width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" class=\\"yfm-clipboard-icon\\" data-animation=\\"9\\">/n
      <path fill=\\"currentColor\\" d=\\"M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z\\">
      </path>/n
      <path stroke=\\"currentColor\\" fill=\\"transparent\\" stroke-width=\\"1.5\\" d=\\"M9.5 13l3 3l5 -5\\" visibility=\\"hidden\\">/n
      <animate id=\\"visibileAnimation-9\\" attributeName=\\"visibility\\" from=\\"hidden\\" to=\\"visible\\" dur=\\"0.2s\\" fill=\\"freeze\\" begin>
      </animate>/n
      <animate id=\\"hideAnimation-9\\" attributeName=\\"visibility\\" from=\\"visible\\" to=\\"hidden\\" dur=\\"1s\\" begin=\\"visibileAnimation-9.end+1\\" fill=\\"freeze\\">
      </animate>/n
      </path>/n
      </svg>/n
      </button>/n
      </div>/n
      <p>
      <code>[not a link](./path/to/relative/file.md)
      </code>
      </p>/n
      <p>
      <a href=\\"mermaid.html\\">Mermaid usage
      </a>
      </p>/n
      <p>
      <a href=\\"mermaid.html#info\\">Mermaid info
      </a>
      </p>/n
      <p>
      <a href=\\"latex.html\\">Latex usage
      </a>
      </p>/n
      <p>Bob
      </p>/n","leading":false},"router":{"pathname":"1","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: 1.md 1`] = `
"# Title

Text not_var{{not-var}} not_var{{skip()}} default-var-value

Text \`not_var{{not-var}}\` \`not_var{{skip()}}\` \`default-var-value\`

Text
\`\`\`
not_var{{not-var}}

not_var{{skip()}}

default-var-value
\`\`\`


\`[not a link](./path/to/relative/file.md)\`

[{#T}](./mermaid.md)

[{#T}](./mermaid.md#info)

[](./latex.md)

{% include [user](includes/user.md) %}
"
`;

exports[`Regression internal: filelist 1`] = `
"[
  "1.md",
  "_assets/1.png",
  "_assets/2.png",
  "_assets/3.png",
  "_assets/empty.pdf",
  "images.md",
  "included-item.md",
  "includes/user.md",
  "index.yaml",
  "latex.md",
  "merge/merge.md",
  "merge/merged.md",
  "merge/toc.yaml",
  "mermaid.md",
  "sub/folder/item-1.md",
  "toc.yaml"
]"
`;

exports[`Regression internal: filelist 2`] = `
"[
  "1.html",
  "_assets/1.png",
  "_assets/2.png",
  "_assets/3.png",
  "_assets/empty.pdf",
  "_bundle/search-async-0",
  "_bundle/app-css-1",
  "_bundle/app-js-1",
  "_bundle/search-css-2",
  "_bundle/latex-extension.css",
  "_bundle/latex-extension.js",
  "_bundle/mermaid-extension.js",
  "_bundle/search-js-0",
  "_bundle/search-js-1",
  "_bundle/search-css-0",
  "_bundle/search-js-2",
  "_bundle/search-css-1",
  "images.html",
  "included-item.html",
  "index.html",
  "latex.html",
  "merge/merge.html",
  "merge/merged.html",
  "merge/toc.js",
  "mermaid.html",
  "sub/folder/item-1.html",
  "toc.js"
]"
`;

exports[`Regression internal: images.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Images
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{},"assets":[],"headings":[],"title":"Images","includes":[],"html":"
      <p>
      <img src=\\"_assets/1.png\\" alt=\\"simple image\\" />
      </p>/n![with leading comment](_assets/2.png)/n
      <p>
      <img src=\\"_assets/3.png\\" alt=\\"with trailing comment\\" />
      </p>/n/n","leading":false},"router":{"pathname":"images","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: images.md 1`] = `
"# Images

![simple image](_assets/1.png)

<!-- comment -->![with leading comment](_assets/2.png)

![with trailing comment](_assets/3.png)<!-- comment -->

<!-- ![commented](_assets/4.png) -->
"
`;

exports[`Regression internal: included-item.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Included item
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{},"assets":[],"headings":[],"title":"Included item","includes":[],"html":"
      <p>Included item text
      </p>/n/n
      <div class=\\"yfm-note yfm-accent-alert\\" note-type=\\"alert\\">
      <p class=\\"yfm-note-title\\">Alert!
      </p>/n
      <div class=\\"yfm-note-content\\">  /n
      </div>
      </div>
      <p>
      <img src=\\"_assets/empty.pdf\\" alt=\\"pdf\\" />
      </p>/n","leading":false},"router":{"pathname":"included-item","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: included-item.md 1`] = `
"# Included item

Included item text

<!-- :   {% include [rec-off](./commented-include.md) %} -->

{% note alert "Alert!" %}

  <!-- content -->

{% endnote %}

![pdf](_assets/empty.pdf)
"
`;

exports[`Regression internal: includes/user.md 1`] = `
"Bob
"
`;

exports[`Regression internal: index.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"data":{"links":[{"name":"Link 1","href":"1.html"},{"name":"Link 2","href":"1/index.html"}],"meta":{}},"meta":{},"title":"","leading":true},"router":{"pathname":"index","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: index.yaml 1`] = `
"links:
  - name: Link 1
    href: ./1.md
  - name: Link 2
    href: ./1
meta: {}
"
`;

exports[`Regression internal: latex.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Latex usage
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
    <link rel="stylesheet"
          type="text/css"
          href="_bundle/latex-extension.css"
          id="custom-style"
    >
    <script src="_bundle/latex-extension.js">
    </script>
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{"style":["_bundle/latex-extension.css"],"script":["_bundle/latex-extension.js"]},"assets":[],"headings":[],"title":"Latex usage","includes":[],"html":"
      <p>
      <span class=\\"yfm-latex\\" data-content=\\"%5Csqrt%7B3x-1%7D%2B(1%2Bx)%5E2\\" data-options=\\"%7B%22displayMode%22%3Afalse%7D\\">
      </span>
      </p>/n","leading":false},"router":{"pathname":"latex","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: latex.md 1`] = `
"# Latex usage

$/sqrt{3x-1}+(1+x)^2$
"
`;

exports[`Regression internal: merge/merge.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="../">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Should use merged item
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{},"assets":[],"headings":[],"title":"Should use merged item","includes":[],"html":"
      <p>
      <a href=\\"merge/merged.html\\">merged
      </a>
      </p>/n","leading":false},"router":{"pathname":"merge/merge","depth":2},"lang":"ru","langs":["ru"]};
    </script>
    <script src="merge/toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: merge/merge.md 1`] = `
"# Should use merged item

[merged](./merged.md)
"
`;

exports[`Regression internal: merge/merged.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="../">
    <meta name="sourcePath"
          content="merge-source/merged.md"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Merged item
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{"sourcePath":"merge-source/merged.md"},"assets":[],"headings":[],"title":"Merged item","includes":[],"html":"
      <p>Should be linked in toc.yaml
      </p>/n","leading":false},"router":{"pathname":"merge/merged","depth":2},"lang":"ru","langs":["ru"]};
    </script>
    <script src="merge/toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: merge/merged.md 1`] = `
"---
sourcePath: merge-source/merged.md
---
# Merged item

Should be linked in toc.yaml
"
`;

exports[`Regression internal: merge/toc.js 1`] = `"window.__DATA__.data.toc = {"items":[{"name":"Use merged","href":"merge/merge.html","id":"UUID"},{"name":"Merged item","href":"merge/merged.html","id":"UUID"}],"id":"UUID"};"`;

exports[`Regression internal: merge/toc.yaml 1`] = `
"items:
  - name: Use merged
    href: merge.md
  - name: Merged item
    href: merged.md
"
`;

exports[`Regression internal: mermaid.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="./">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Mermaid usage
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
    <script src="_bundle/mermaid-extension.js">
    </script>
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{"script":["_bundle/mermaid-extension.js"]},"assets":[],"headings":[{"title":"Mermaid info","href":"#info","level":2}],"title":"Mermaid usage","includes":[],"html":"
      <div class=\\"mermaid\\" data-content=\\"sequenceDiagram%0A%20%20%20%20%20%20%20%20rect%20rgba(251%2C%20243%2C%20232%2C%200.2)%0A%20%20%20%20%20%20%20%20%20%20%20%20Alice%20-%3E%3E%20Bob%3A%0A%20%20%20%20%20%20%20%20end%0A\\">
      </div>
      <h2 id=\\"info\\">
      <a href=\\"#info\\" class=\\"yfm-anchor\\" aria-hidden=\\"true\\">
      <span class=\\"visually-hidden\\" data-no-index=\\"true\\">Mermaid info
      </span>
      </a>Mermaid info
      </h2>/n
      <p>Some mermaid info
      </p>/n","leading":false},"router":{"pathname":"mermaid","depth":1},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: mermaid.md 1`] = `
"# Mermaid usage

\`\`\`mermaid

  sequenceDiagram
        rect rgba(251, 243, 232, 0.2)
            Alice ->> Bob:
        end
  \`\`\`

## Mermaid info {#info}

Some mermaid info
"
`;

exports[`Regression internal: sub/folder/item-1.html 1`] = `
<!DOCTYPE html>
<html lang="ru"
      dir="ltr"
>
  <head>
    <meta charset="utf-8">
    <base href="../../">
    <meta name="viewport"
          content="width=device-width, initial-scale=1.0"
    >
    <title>
      Item 1
    </title>
    <style type="text/css">
      body {
                        height: 100vh;
                    }
    </style>
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/search-css-0"
    >
    <link type="text/css"
          rel="stylesheet"
          href="_bundle/app-css-1"
    >
  </head>
  <body class="g-root g-root_theme_light">
    <div id="root">
    </div>
    <script type="application/javascript">
      window.STATIC_CONTENT = false
                   window.__DATA__ = {"data":{"meta":{},"assets":[],"headings":[],"title":"Item 1","includes":[],"html":"
      <p>Item 1 text
      </p>/n","leading":false},"router":{"pathname":"sub/folder/item-1","depth":3},"lang":"ru","langs":["ru"]};
    </script>
    <script src="toc.js"
            type="application/javascript"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-0"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/app-js-1"
    >
    </script>
    <script type="application/javascript"
            src="_bundle/search-js-2"
    >
    </script>
  </body>
</html>
`;

exports[`Regression internal: sub/folder/item-1.md 1`] = `
"# Item 1

Item 1 text
"
`;

exports[`Regression internal: toc.js 1`] = `"window.__DATA__.data.toc = {"items":[{"name":"Verbose root (index.yaml) will be transformed to index.html","href":"index.html","id":"UUID"},{"name":"Root will be transformed to index.html","href":"index.html","id":"UUID"},{"name":"Md item with not_var syntax","href":"1.html","id":"UUID"},{"name":"Md item named without extension","href":"1.html","id":"UUID"},{"name":"Item with empty href","id":"UUID"},{"name":"Included Item","href":"included-item.html","id":"UUID"},{"name":"Named include (items is Object here - this is not an error)","items":[{"name":"Item 1","href":"sub/folder/item-1.html","id":"UUID"}],"id":"UUID"},{"name":"Mermaid","href":"mermaid.html","id":"UUID"},{"name":"Latex","href":"latex.html","id":"UUID"},{"name":"Images","href":"images.html","id":"UUID"}],"id":"UUID"};"`;

exports[`Regression internal: toc.yaml 1`] = `
"items:
  - name: Verbose root (index.yaml) will be transformed to index.html
    href: index.yaml
  - name: Root will be transformed to index.html
    href: /index.yaml
  - name: Md item with not_var syntax
    href: 1.md
  - name: Md item named without extension
    href: 1.md
  - name: Item with empty href
  - name: Included Item
    href: included-item.md
  - name: Named include (items is Object here - this is not an error)
    items:
      - name: Item 1
        href: sub/folder/item-1.md
  - name: Mermaid
    href: mermaid.md
  - name: Latex
    href: latex.md
  - name: Images
    href: images.md
"
`;
